Recon/src/Recon/Skims/RecoilFiducialityProcessor.cxx

```diff
@@ -0,0 +1,143 @@
+/**
+ * @file RecoilFiducialityProcessor.cxx
+ * @brief Processor used to select events where the recoil electron is fiducial.
+ * @author Elizabeth Berzin, Stanford University
+ */
+
+#include "Recon/Skims/RecoilFiducialityProcessor.h"
+
+namespace recon {
+
+void RecoilFiducialityProcessor::configure(
+    framework::config::Parameters &parameters) {
+  min_p_mag_ = parameters.getParameter<double>("min_p_mag");
+  min_tracker_hits_ = parameters.getParameter<int>("min_tracker_hits");
+  ecal_collection_ = parameters.getParameter<std::string>("ecal_collection");
+  hcal_collection_ = parameters.getParameter<std::string>("hcal_collection");
+  recoil_collection_ =
+      parameters.getParameter<std::string>("recoil_collection");
+  output_collection_ =
+      parameters.getParameter<std::string>("output_collection");
+  inverse_skim_ = parameters.getParameter<bool>("inverse_skim");
+}
+
+void RecoilFiducialityProcessor::produce(framework::Event &event) {
+  // Get the collection of simulated particles from the event
+  auto particleMap{event.getMap<int, ldmx::SimParticle>("SimParticles")};
+
+  // Search for the recoil electron
+  auto [recoil_track_id, recoil_electron] = Analysis::getRecoil(particleMap);
+
+  // Get the collection of simulated Ecal hits from the event.
+  const std::vector<ldmx::SimCalorimeterHit> ecal_sim_hits =
+      event.getCollection<ldmx::SimCalorimeterHit>(ecal_collection_);
```

Comment:
please add pass name here and to all other collections + to the configuration and python config (defaulting to an empty string). selecting among several passes is a very useful capability that all processors should have, imo